frontend/src/components/UI/DocumentCard/DocumentCard.jsx

```diff
@@ -0,0 +1,50 @@
+import PropTypes from "prop-types";
+import cssClasses from "./styles.module.css";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faEye } from "@fortawesome/free-solid-svg-icons";
+/**
+ * Renders an individual document
+ * @component
+ * @param {Object} props - The component props.
+ * @param {string} [props.className] - Additional CSS classes.
+ * @param {string} [props.dataCy] - Data attribute for cypress tests
+ * @param {Object} props.document - The document this card renders
+ * @returns {JSX.Element} - The rendered component
+ */
+const DocumentCard = ({ className, dataCy, document, ...rest }) => {
+    const cardContainer = `bg-base-lightest border-base-lighter font-family-sans display-flex ${cssClasses.container} ${className}`;
+    const icon_class_names = `${cssClasses.eyeIcon} height-3 width-3`;
+    const documentTitleClassNames = `${cssClasses.documentTitle} font-sans-3xs`;
+    const uploaded_by_text = `Uploaded by ${document.uploaded_by} on ${document.upload_date}`;
+    return (
+        <div
+            className={cardContainer}
+            data-cy={dataCy ?? dataCy}
+            {...rest}
+        >
+            <div className="display-flex flex-column margin-left-4 margin-top-205">
+                <div className={documentTitleClassNames}>{document.title}</div>
+                <div className="font-sans-3xs text-bold margin-top-05">{document.filename}</div>
```

Comment:
to get the 12px font-size we came up with a custom classname called 'font-12px' 😄

Reply:
Thanks, I did not know this. Also I had similar feelings about potentially moving the documents into components as well. I will do that.